### ISSUE TITLE
OSD not being created with "lsblk: /dev/sdc: not a block device" error

### ISSUE THREAD

**Issue by leopaul36** [rook/rook#4967]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:** OSD not being created with "lsblk: /dev/sdc: not a block device" error

**Expected behavior:** OSD being created successfully

**How to reproduce it (minimal and precise):** `kubectl create -f my-cluster.yml`

**File(s) to submit**:

* `my-cluster.yaml`:
```
apiVersion: ceph.rook.io/v1
kind: CephCluster
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  cephVersion:
    # For the latest ceph images, see https://hub.docker.com/r/ceph/ceph/tags
    image: ceph/ceph:v14.2.6
  dataDirHostPath: /var/lib/rook
  mon:
    count: 3
  dashboard:
    enabled: true
  storage:
    useAllNodes: true
    useAllDevices: false
    nodes:
      - name: "10.0.0.70"
        devices:
        - name: "sdc"
      - name: "10.0.0.71"
        devices:
        - name: "sdc"
      - name: "10.0.0.72"
        devices:
        - name: "sdc" 
```
* rook-ceph-osd-prepare logs:

```
2020-03-05 17:27:47.397513 I | cephcmd: desired devices to configure osds: [{Name: OSDsPerDevice:1 MetadataDevice: DatabaseSizeMB:0 DeviceClass: IsFilter:false IsDevicePathFilter:false}]
2020-03-05 17:27:47.400870 I | rookcmd: starting Rook v1.1.0-beta.0.989.g022ff8a with arguments '/rook/rook ceph osd provision'
2020-03-05 17:27:47.400895 I | rookcmd: flag values: --cluster-id=b7305a33-d8b2-4eb1-9a16-8a4e8b49c878, --data-device-filter=, --data-device-path-filter=, --data-devices=, --encrypted-device=false, --force-format=false, --help=false, --location=, --log-flush-frequency=5s, --log-level=INFO, --metadata-device=, --node-name=k8s-1, --operator-image=, --osd-database-size=0, --osd-store=, --osd-wal-size=576, --osds-per-device=1, --pvc-backed-osd=false, --service-account=
2020-03-05 17:27:47.400899 I | op-mon: parsing mon endpoints: a=10.98.207.55:6789,b=10.109.123.29:6789,c=10.104.59.91:6789
2020-03-05 17:27:47.413468 I | op-osd: CRUSH location=root=default host=k8s
2020-03-05 17:27:47.413493 I | cephcmd: crush location of osd: root=default host=k8s
2020-03-05 17:27:47.423968 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2020-03-05 17:27:47.424030 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2020-03-05 17:27:47.424114 I | cephosd: discovering hardware
2020-03-05 17:27:47.424122 I | exec: Running command: lsblk --all --noheadings --list --output KNAME
2020-03-05 17:27:47.430694 I | exec: Running command: lsblk /dev/loop1 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-03-05 17:27:47.433067 W | inventory: skipping device "loop1" because 'lsblk' failed. Failed to complete 'lsblk /dev/loop1': exit status 1. lsblk: /dev/loop1: not a block device
2020-03-05 17:27:47.433092 I | exec: Running command: lsblk /dev/nbd3 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-03-05 17:27:47.434747 W | inventory: skipping device "nbd3" because 'lsblk' failed. Failed to complete 'lsblk /dev/nbd3': exit status 1. lsblk: /dev/nbd3: not a block device
2020-03-05 17:27:47.460767 I | exec: Running command: lsblk /dev/sdc --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-03-05 17:27:47.462273 W | inventory: skipping device "sdc" because 'lsblk' failed. Failed to complete 'lsblk /dev/sdc': exit status 1. lsblk: /dev/sdc: not a block device
```

What bothers me is thre `lsblk: /dev/XXX: not a block device` while looping through every device. From what I have found in successful logs, it should tell me that the device has a filesystem, cannot be used, etc.

I feel like the container responsible for preparing the OSD does not have the privilege to run lsblk on the devices.

Did I miss something in the pre-requisites ?

Thanks in advance for your help.

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 19.04
* Kernel (e.g. `uname -a`): 5.0.0-38-generic
* Cloud provider or hardware configuration: Bare Metal VM
* Rook version (use `rook version` inside of a Rook Pod): rook: v1.1.0-beta.0.989.g022ff8a
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`): v1.17.3

**Comment by travisn** [comment]
@leopaul36 Looks like this issue is in the master builds. Until we get this fixed, please use the latest stable release [v1.2.5](https://github.com/rook/rook/tree/v1.2.5/cluster/examples/kubernetes/ceph).

If you run `kubectl describe pod` on the osd prepare pod, the `/dev` mount is likely missing.

**Comment by leopaul36** [comment]
Hi @travisn,

Indeed, the /dev is missing from the mounts in the prepare pods:

```
    Mounts:
      /etc/ceph from ceph-conf-emptydir (rw)
      /rook from rook-binaries (rw)
      /var/lib/ceph/crash from rook-ceph-crash (rw)
      /var/lib/rook from rook-data (rw)
      /var/log/ceph from rook-ceph-log (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from rook-ceph-osd-token-q4t5w (ro)
```

Although I tried to re-create both `common.yaml` and `operator.yaml` as well as the `my-cluster.yaml` from the v1.2.5 release and again from the `always-mount-dev` branch on your fork but I still get the same behavior...

Not sure If it is related but I noticed this at the end of the describe of the prepare pod:

```
Warning  FailedCreatePodSandBox  8m17s  kubelet, k8s-1  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "rook-ceph-osd-prepare-k8s-1-kzw8n": Error response from daemon: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:438: writing syncT 'resume' caused \\\"write init-p: broken pipe\\\"\"": unknown
```

I feel like there is something wrong with my docker, I will try to look into that.

Thanks for your help.
